File: 2550-words-within-two-edits-of-dictionary/words-within-two-edits-of-dictionary.java
class Solution{
    void solve(String w, String s, int[] t){
        for(int i=0; i<w.length(); i++){
            if(w.charAt(i) != s.charAt(i))
                t[0]++;
        }
    }

    public List<String> twoEditWords(String[] q, String[] d){
        int n=q.length;
        int m=d.length;

        List<String> res=new ArrayList<>();
        for(String it: q){
            String w=it;
            for(int i=0; i<m; i++){
                int[] t=new int[1];
                String s=d[i];
                solve(w, s, t);
                if(t[0]<=2){
                    res.add(w);
                    break;
                }
            }
        }

        return res;
    }
}

File: 2550-words-within-two-edits-of-dictionary/words-within-two-edits-of-dictionary.cpp
class Solution{
public:
    void solve(string &w, string &s, int &t){
        for(int i=0; i<w.size(); i++){
            if(w[i] != s[i])
                t++;
        }
    }

    vector<string> twoEditWords(vector<string> &q, vector<string> &d){
        int n=q.size();
        int m=d.size();

        vector<string> res;
        for(auto it: q){
            string w=it;
            for(int i=0; i<m; i++){
                int t=0;
                string s=d[i];
                solve(w, s, t);
                if(t<=2){
                    res.push_back(w);
                    break;
                }
            }
        }

        return res;

    }
};